src/gc/registry.d

```diff
@@ -0,0 +1,87 @@
+/**
+ * Contains a registry for GC factories.
```

Comment:
I'm slightly confused by the terminology here: I would have expected the "registry" to be the "factory" and a GC is registered by adding a "factory method" to the factory.

Reply:
Well, yes it's a registry for factory methods.